src/analysis/souffle/dataflow_graph.dl

```diff
@@ -73,4 +89,21 @@ path(from, to) :- resolvedEdge(_, from, intermediate), path(intermediate, to).
 isAccessPath(x) :- resolvedEdge(_, x, _).
 isAccessPath(y) :- resolvedEdge(_, _, y).
 
+// The 0th predecessor shall be the predecessor with the smallest src ordinal
+// number.
+orderedPredecessors(ap, for, src, 0) :-
+  resolvedEdge(for, src, ap),
+  ord(src) = min ord(candidate) : { resolvedEdge(for, candidate, ap) }.
```

Comment:
Is `ord` a souffle primitive?

Reply:
Yes. `ord` is the [ordinal number for the symbol](https://souffle-lang.github.io/types#symbol-type) that Souffle uses internally. The documentation used to have language that implied greater degrees about stability (ie, `ord` would be assigned to symbols in the order that Souffle saw them) but now basically says that `ord` will be an arbitrary unique mapping between symbols and numbers.

It's less stable than I'd like and very Souffle-specific. In the future, if we find that it matters, we can have a more stable mapping from `AccessPath`s to unique IDs by just having the IR translator output a unique ID for every `AccessPath` produced. If we switch to representing `AccessPath`s with ADTs, we will need to do that, as Souffle only produces `ord` numbers for symbols. But I think `ord` works for now.